app/packages/looker/src/elements/index.ts

```diff
@@ -10,26 +10,25 @@ import {
   ThreeDState,
   VideoState,
 } from "../state";
+import type { BaseElement } from "./base";
 import * as common from "./common";
 import * as frame from "./frame";
 import * as image from "./image";
 import * as imavid from "./imavid";
 import * as pcd from "./three-d";
+import type { ElementsTemplate } from "./util";
 import { createElementsTree, withEvents } from "./util";
 import * as video from "./video";
 
-export type GetElements<State extends BaseState> = (
-  config: Readonly<State["config"]>,
-  update: StateUpdate<State>,
-  dispatchEvent: (eventType: string, details?: any) => void,
-  batchUpdate?: (cb: () => unknown) => void | undefined
-) => common.LookerElement<State>;
+export type GetElements<State extends BaseState> = (params: {
+  abortController: AbortController;
+  batchUpdate?: (cb: () => unknown) => void;
+  config: Readonly<State["config"]>;
+  dispatchEvent: (eventType: string, details?: any) => void;
+  update: StateUpdate<State>;
+}) => common.LookerElement<State>;
 
-export const getFrameElements: GetElements<FrameState> = (
-  config,
-  update,
-  dispatchEvent
-) => {
+export const getFrameElements: GetElements<FrameState> = (params) => {
```

Comment:
[nit] do we know the expected type of `params` for these methods?

Reply:
The `GetElements` function type defines `params`